src/erlfdb.erl

```diff
@@ -1903,6 +1911,15 @@ do_transaction(?IS_TX = Tx, UserFun) ->
             put(?ERLFDB_ERROR, Code),
             wait(on_error(Tx, Code), [{timeout, infinity}]),
             do_transaction(Tx, UserFun)
+    after
```

Comment:
This is a bit awkward as it will run for each do_transaction. If the UserFun is run more than twice, the flush runs more than once. We could introduce a `Depth` var in do_transaction that would allow us to ensure the flush only runs once, but increases the code complexity a bit.

Reply:
After thinking about this a bit more, using an `after` like this will blow out the stack frame. We should move the logic into the catch and we will have to introduce a `Depth` to ensure all other recursive `do_transaction`s remain tail calls.